### PR TITLE
[Go] Move rakefile out

### PR DESCRIPTION
**Pull request by mcardosos**
Currently, we use the rakefile to generate the SDK (only management plane) with a single command: `rake SDK_VERSION=3.14.16`. With this change, we could also generate dataplane.
 Something important about this PR is...
@jhendrixMSFT and I would like dataplane packages to be included in their own folder in the [Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go), just like arm packages are structured today. Maybe `dataplane` is not the cutest name for the folder, we can discuss that here too.

- arm
 - authorization
 - batch
 - cdn
 - etc
- dataplane
 - batch
 - insights
 - keyvault
 - etc
- management (asm)

Please take a look @jhendrixMSFT @garimakhulbe @vishrutshah

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mcardosos__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mcardoso). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by garimakhulbe** [comment]
LGTM. Thank you.

**Comment by tbombach** [comment]
@azuresdkci add to whitelist

**Comment by garimakhulbe** [comment]
@mcardosos Where are you planning to keep data plane APIs in sdk? Have you discussed with team?

**Comment by azurecla** [comment]
Hi __@mcardosos__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mcardoso). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by garimakhulbe** [comment]
@mcardosos closed it by mistake. reopened. :)
Have you discussed with other SDK teams? How they are handling this?

**Comment by tbombach** [comment]
Are there plans to move this file out of the AutoRest repo? This kind of configuration file should live in the Go language SDK, not AutoRest (see a similar file in Node: https://github.com/Azure/azure-sdk-for-node/blob/master/swagger_to_sdk_config.json)

**Comment by mcardosos** [comment]
@garimakhulbe I have not discussed it with other teams. Just with @jhendrixMSFT and @vishrutshah
@tbombach I haven't thought about moving this file. I will modify it so it can live with the SDK.

**Comment by jhendrixMSFT** [comment]
@mcardosos There was a suggestion on another PR to use the acronym "adp" for the name of the data plane directory.  It's short and consistent with the other naming conventions.  Thoughts?

File: src/generator/AutoRest.Go/rakefile.rb [hunk 1]
```diff
@@ -8,18 +8,19 @@
...
 	cognitiveservices: {version: "2016-02-01-preview"},
-	compute: {version: "2016-03-30"},
+	compute: {version: "2016-03-30"}, #composite swagger (includes container service)
 	# containerservice: {version: "2016-03-30", swagger: "containerservice"},
 	# commerce: {version: "2015-06-01-preview"},
 	datalake_analytics: {
 		account: {version: "2015-10-01-preview"},
 		# catalog: {version: "2016-06-01-preview"},
-		job:  {version: "2016-03-20-preview"}
+		# job:  {version: "2016-03-20-preview"}
```
Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/rakefile.rb]: Why is this one disabled?

File: src/generator/AutoRest.Go/rakefile.rb [hunk 2]
```diff
@@ -211,30 +240,30 @@ def generate(service)
...
 	s = `#{ENV["GOPATH"]}\\bin\\golint #{service.namespace}`
 	print s
 end
 
 def vet(service)
-	puts "Vetting #{service.fullname}"
+	puts "Vetting #{service.plane} #{service.fullname}"
 	s = `go vet #{service.namespace}`
 	print s
 end
```
Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/rakefile.rb]: nit: add a new-line at the end of the file.

File: src/generator/AutoRest.Go/rakefile.rb (hunk 1 above)
Comment (mcardosos) [inline src/generator/AutoRest.Go/rakefile.rb]: If I try to generate this one, the result is a bit ugly. And we are not generating job right now. In the other hand, these are dataplane.

**Comment by mcardosos** [comment]
I like that "adp" is short, but maybe at first glance it is not so obvious it refers to dataplane.

**Comment by garimakhulbe** [comment]
I have no clue on this but between `adp` and `dataplane`, I will pick `dataplane`. 

In import..
`github.com/azure/azure-sdk-for-go/adp/insights-... `
`github.com/azure/azure-sdk-for-go/dataplane/insights-....` // infer easily.

**Comment by jhendrixMSFT** [comment]
My only concern was if "dataplane" is internal terminology which doesn't seem to be the case.  I believe that @mcardosos was going to follow up with some of the tier-one langues to see how they named dataplane content.

**Comment by mcardosos** [comment]
[Java](https://github.com/Azure/azure-sdk-for-java), [Python](https://github.com/Azure/azure-sdk-for-python) and [Node](https://github.com/Azure/azure-sdk-for-node/tree/master/lib/services) is following this structure (somewhat similar to what we have now with storage dataplane, but we place all mgmt in arm folder)
- azure-batch
- azure-keyvault
- azure-mgmt-batch
- azure-mgmt-compute
- azure-mgmt-graph-rbac
- azure-mgmt-keyvault

[.Net](https://github.com/Azure/azure-sdk-for-net/tree/AutoRest/src) sometimes places dataplane with management, but not always.
None use dataplane as a folder name. We could just place all dataplane in the root, but I think it is more organized with a folder.

**Comment by mcardosos** [comment]
Moved (and translated to golang) rakefile to Go SDK repo :3

**Comment by mcardosos** [comment]
@azuresdkci retest this please!

**Pull request merged**